src/Apps/Consign/Hooks/useSubmissionFlowSteps.ts

```diff
@@ -0,0 +1,35 @@
+import { themeProps } from "@artsy/palette"
+import { useFeatureFlag } from "System/useFeatureFlag"
+import { __internal__useMatchMedia } from "Utils/Hooks/useMatchMedia"
+
+function typedArray<T extends string>(...elems: T[]): T[] {
+  return elems
+}
+
+export const submissionFlowSteps = typedArray(
+  "Artwork Details",
+  "Upload Photos",
+  "Contact Information"
+)
+
+export const submissionFlowStepsMobile = typedArray(
+  "Artwork",
+  "Photos",
+  "Contact"
+)
+
+export const useSubmissionFlowSteps = () => {
+  const enableFlowReorder = useFeatureFlag(
+    "reorder-swa-artwork-submission-flow"
+  )
+  const isMobile = __internal__useMatchMedia(themeProps.mediaQueries.xs)
+  if (enableFlowReorder && isMobile) {
+    return typedArray("Contact", "Artwork", "Photos")
+  } else if (enableFlowReorder) {
+    return typedArray("Contact Information", "Artwork Details", "Upload Photos")
+  }
+  if (isMobile) {
+    return submissionFlowStepsMobile
+  }
+  return submissionFlowSteps
```

Comment:
I would flatten out all these conditionals.

Reply:
Oh, I thought this is as flat as it can be, since I had avoided nesting.

Reply:
Oh, yes you're right of course. The else if threw me. I have a personal preference to never use else or else if in favor of short-circuiting returns. I would prefer that of course. Though that is preference.

I also want to make sure you realize that the `__internal__useMatchMedia(themeProps.mediaQueries.xs)` is not SSR friendly.

Reply:
I mean in this case actually the `else if` doesn't actually do anything. Just make it an if

Reply:
Oh. Yeah true. Will update. 🙌 .
We only use `__internal__useMatchMedia` to determine what labels to put on the Submission steps